Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportInputController.cpp

```diff
@@ -0,0 +1,185 @@
+    }
+
+    AzToolsFramework::ViewportInteraction::KeyboardModifier AnimViewportInputController::GetKeyboardModifier(
+        const AzFramework::InputChannel& inputChannel)
+    {
+        using AzToolsFramework::ViewportInteraction::KeyboardModifier;
+        using Key = AzFramework::InputDeviceKeyboard::Key;
+        const auto& id = inputChannel.GetInputChannelId();
+        if (id == Key::ModifierAltL || id == Key::ModifierAltR)
+        {
+            return KeyboardModifier::Alt;
+        }
+        if (id == Key::ModifierCtrlL || id == Key::ModifierCtrlR)
+        {
+            return KeyboardModifier::Ctrl;
+        }
+        if (id == Key::ModifierShiftL || id == Key::ModifierShiftR)
+        {
+            return KeyboardModifier::Shift;
+        }
+        return KeyboardModifier::None;
+    }
+
+    bool AnimViewportInputController::HandleInputChannelEvent(const AzFramework::ViewportControllerInputEvent& event)
+    {
+        using AzFramework::InputChannel;
+        using AzToolsFramework::ViewportInteraction::KeyboardModifier;
+        using AzToolsFramework::ViewportInteraction::MouseButton;
+        using AzToolsFramework::ViewportInteraction::MouseEvent;
+        using AzToolsFramework::ViewportInteraction::MouseInteraction;
+        using AzToolsFramework::ViewportInteraction::MouseInteractionEvent;
+        using AzToolsFramework::ViewportInteraction::ProjectedViewportRay;
+        using AzToolsFramework::ViewportInteraction::ViewportInteractionRequestBus;
+
+        bool interactionHandled = false;
+        float wheelDelta = 0.0f;
+        AZStd::optional<MouseButton> overrideButton;
+        AZStd::optional<MouseEvent> eventType;
+
+        const auto state = event.m_inputChannel.GetState();
+        if (IsMouseMove(event.m_inputChannel))
+        {
+            const auto* position = event.m_inputChannel.GetCustomData<AzFramework::InputChannel::PositionData2D>();
+            AZ_Assert(position, "Expected PositionData2D but found nullptr");
+
+            AzFramework::WindowSize windowSize;
+            AzFramework::WindowRequestBus::EventResult(
+                windowSize, event.m_windowHandle, &AzFramework::WindowRequestBus::Events::GetClientAreaSize);
+
+            const auto screenPoint = AzFramework::ScreenPoint(
+                aznumeric_cast<int>(position->m_normalizedPosition.GetX() * windowSize.m_width),
+                aznumeric_cast<int>(position->m_normalizedPosition.GetY() * windowSize.m_height));
+
+            ProjectedViewportRay ray{};
+            ViewportInteractionRequestBus::EventResult(
+                ray, GetViewportId(), &ViewportInteractionRequestBus::Events::ViewportScreenToWorldRay, screenPoint);
+
+            m_mouseInteraction.m_mousePick.m_rayOrigin = ray.m_origin;
+            m_mouseInteraction.m_mousePick.m_rayDirection = ray.m_direction;
+            m_mouseInteraction.m_mousePick.m_screenCoordinates = screenPoint;
+
+            eventType = MouseEvent::Move;
+        }
```

Comment:
suggestion: I know some of this is similar to the code in `ManipulatorViewportController` but not all - it could be worth extracting some of this code to functions to make it easier to reuse and be more modular, I'd definitely be in favour of tihs! 🙂